src/algebra/module/basic.lean

```diff
@@ -270,64 +277,31 @@ section add_comm_monoid
 
 variables [semiring R] [add_comm_monoid M] [semimodule R M]
 
-/-- The natural ℕ-semimodule structure on any `add_comm_monoid`. -/
--- We don't make this a global instance, as it results in too many instances,
--- and confusing ambiguity in the notation `n • x` when `n : ℕ`.
-def add_comm_monoid.nat_semimodule : semimodule ℕ M :=
-{ smul := nsmul,
-  smul_add := λ _ _ _, nsmul_add _ _ _,
-  add_smul := λ _ _ _, add_nsmul _ _ _,
-  mul_smul := λ _ _ _, mul_nsmul _ _ _,
-  one_smul := one_nsmul,
-  zero_smul := zero_nsmul,
-  smul_zero := nsmul_zero }
-
-section
-local attribute [instance] add_comm_monoid.nat_semimodule
-/-- `nsmul` is defined as the `smul` action of `add_comm_monoid.nat_semimodule`. -/
-lemma nsmul_def (n : ℕ) (x : M) :
-  n •ℕ x = n • x :=
-rfl
-end
-
 section
 variables (R)
 /-- `nsmul` is equal to any other semimodule structure via a cast. -/
 lemma nsmul_eq_smul_cast (n : ℕ) (b : M) :
-  n •ℕ b = (n : R) • b :=
+  n • b = (n : R) • b :=
 begin
-  rw nsmul_def,
   induction n with n ih,
   { rw [nat.cast_zero, zero_smul, zero_smul] },
-  { rw [nat.succ_eq_add_one, nat.cast_succ, add_smul, add_smul, one_smul, ih, one_smul] }
+  { rw [nat.succ_eq_add_one, nat.cast_succ, add_smul, add_smul, one_smul, ih, one_smul], }
 end
 end
 
-/-- `nsmul` is equal to any `ℕ`-semimodule structure. -/
-lemma nsmul_eq_smul [semimodule ℕ M] (n : ℕ) (b : M) : n •ℕ b = n • b :=
-by rw [nsmul_eq_smul_cast ℕ, n.cast_id]
-
-/-- All `ℕ`-semimodule structures are equal. -/
-instance add_comm_monoid.nat_semimodule.subsingleton : subsingleton (semimodule ℕ M) :=
-⟨λ P Q, by {
-  ext n,
-  rw [←nsmul_eq_smul, ←nsmul_eq_smul], }⟩
```

Comment:
Is there a reason for removing this? It seems like a nice statement to have to justify the design, even if it's not really needed any more.

(obviously let's not stop the build, but I think we should add this back)